Minor improvements to README and shell.nix

delan opened the pull request:
While testing this, I ran into a few minor issues:

- the demo can only be built in release mode, due to [servo/servo#31059](https://github.com/servo/servo/issues/31059)
- the build takes over 11 minutes with gnumake 4.4, due to [servo/mozjs#375](https://github.com/servo/mozjs/issues/375)
- the demo panics unless I switch to Wayland, because I normally use X11 (“Failed to bootstrap native connection: NoCurrentConnection” at src/renderer.rs:267)

This patch adds Servo’s workarounds for the first two in shell.nix, and documents the third in the README.

delan committed:
Minor improvements to README and shell.nix

vimpostor commented on shell.nix:
```diff
@@ -6,10 +6,11 @@
+	stdenv = stdenvAdapters.useMoldLinker llvmPackages.stdenv;
 in
-	stdenv.mkDerivation {
+	llvmPackages.stdenv.mkDerivation {
```
Thanks very much for your PR and for tracking down all these upstream issues :tada: 

The changes look mostly good to me, but is there a reason you resort back to `llvmPackages.stdenv` here instead of using just `stdenv`, which was defined as `stdenvAdapters.useMoldLinker llvmPackages.stdenv`, i.e. you are effectively removing mold as linker here?

delan replied:
thanks! this was a mistake, i intended to prepend it to the stdenv in the definition only.

delan replied:
fixed in 64419d9909837bcad5b2b626ac74f1134dffc137

delan committed:
Fix typo in shell.nix

vimpostor committed:
Explicitly set LIBCLANG_PATH

Some dependencies (e.g. mozjs_sys) need it to find clang, otherwise it
tries to use the system clang:
```
Unable to find libclang: "the `libclang` shared library at /usr/lib/libclang.so.17.0.6 could not be opened: libLLVM-17.so: cannot open shared object file: No such file or directory"
```

The relevant check happens in clang-sys [0] and unfortunately we cannot
use its llvm-config path on Nix (--prefix does not work), so we have to
resort to setting LIBCLANG_PATH directly.

We add llvm-config anyway via llvmPackages.llvm, as it is helpful to
have when used with some other options (e.g. --libdir, which **does**
work on Nix).

[0] https://github.com/KyleMayes/clang-sys/blob/99f4007e0b304631850f34cc1be89ac74f593ab8/build/common.rs#L287

vimpostor commented on shell.nix:
```diff
@@ -6,8 +6,9 @@
-	stdenv = stdenvAdapters.useMoldLinker clangStdenv;
+	pkgs_gnumake_4_3 = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/6adf48f53d819a7b6e15672817fa1e78e5f4e84f.tar.gz") {};
+	llvmPackages = llvmPackages_14; # servo/servo#31059
+	stdenv = stdenvAdapters.useMoldLinker llvmPackages.stdenv;
```
Hmm, the change from `clangStdenv` to `llvmPackages.stdenv` somehow causes the build to fail for me in mozsj_sys now, as for some reason it attemps to use my system clang:
```
  Unable to find libclang: "the `libclang` shared library at /usr/lib/libclang.so.17.0.6 could not be opened: libLLVM-17.so: cannot open shared object file: No such file or directory"
```

I played around with adding `llvmPackages.llvm` to `buildInputs`, because [clang-sys uses llvm-config --prefix](https://github.com/KyleMayes/clang-sys/blob/99f4007e0b304631850f34cc1be89ac74f593ab8/build/common.rs#L287) to find out the clang location, but unfortunately that doesn't work (probably because there isn't a single derivation that contains the whole clang prefix).

Instead I added a commit now to set `LIBCLANG_PATH` directly, which got things working again.

Thanks for your PR. :)

> Note that this demo currently supports Wayland only, not X11.

FYI there already is a WIP to fix that ;)
